src/node/build/index.ts

```diff
@@ -411,7 +412,10 @@ async function doBuild(options: Partial<BuildConfig>): Promise<BuildResult[]> {
   }
 
   const outDir = path.resolve(root, config.outDir)
-  const indexPath = path.resolve(root, 'index.html')
+  const indexPath = path.resolve(
+    root,
+    entry.endsWith('.html') ? entry : 'index.html'
```

Comment:
Why do we need to check the extension?

Reply:
It maybe passed js file as entry..

Reply:
I'd say it's the developers' duty to find out if they have passed the wrong file. File extensions do not guarantee their content anyway, I would prefer to leave the flexibility to users here.

Reply:
@antfu The `indexPath` must never be a JS module, yet `entry` is allowed to be a JS module (I think that's why it was added, for situations where `index.html` is handcrafted). Checking the extension is the best solution, IMO.